tests/unit/runtimes/node/in_source_config.test.ts

```diff
@@ -285,6 +286,17 @@ describe('V2 API', () => {
 
       expect(isc).toEqual({ inputModuleFormat: 'cjs', routes: [], runtimeAPIVersion: 2 })
     })
+
+    test('ESM file with a default export consisting of a function call', () => {
+      const source = `import * as build from "@remix-run/dev/server-build";
+      import { createRequestHandler } from "@netlify/remix-edge-adapter";
+      
+      export default createRequestHandler({ build });`
+
+      const isc = parseSource(source, options)
+
+      expect(isc).toEqual({ inputModuleFormat: 'esm', routes: [], runtimeAPIVersion: 2 })
```

Comment:
@ascorbic I've used the source you shared in your example and it seems to be detected as a v2 function. Are you sure the detection is what's failing in your case?